docs/user_utilities.rst

```diff
@@ -0,0 +1,154 @@
+
+    - The default logging level is ``ti.INFO``.You can also override the default logging level by setting the environment variable like ``TI_LOG_LEVEL=warn``.
+
+.. function:: ti.info(info)
+
+    :parameter info: (string) logging info
+
+    Print out the input string in green color to stdout, when the logging level is lower or equal to ``ti.INFO``. For example:
+
+.. code-block:: python
+
+    import taichi as ti
+    ti.init(arch = ti.cpu)
+
+    ti.set_logging_level(ti.INFO)
+    var = ti.var(ti.f32, shape = 1)
+
+    @ti.kernel
+    def compute():
+        var[0] = 1.0
+        ti.info(f"set var[0] = 1.0")
+
+    compute()
+
+.. function:: ti.warn(info)
+
+    :parameter info: (string) logging info
+
+    Print out the input string in yellow color to stdout, when the logging level is lower or equal to ``ti.WARN``.
+
+.. function:: ti.debug(info)
+
+    :parameter info: (string) logging info
+
+    Print out the input string in blue color to stdout, when the logging level is lower or equal to ``ti.DEBUG``.
+
+.. function:: ti.trace(info)
+
+    :parameter info: (string) logging info
+
+    Print out the input string in blue color to stdout, only when the logging level is equal to ``ti.TRACE``.
+
+.. function:: ti.error(info)
+
+    :parameter info: (string) logging info
+
+    This function prints out the input string in red color to stdout in any logging level.
+
+    Note that it crashes the Taichi kernel as well as your Taichi program then throw an ``RuntimeError`` out.
+
+.. Warn::
+
+    The usages of ``ti.trace()``, ``ti.debug()`` and ``ti.warn()`` are all similar to ``ti.info()``.But ``ti.error()`` is different from them. When it is called, the program will be crashed immediately.
+
+    In spite of this, the output of ``ti.error()`` is still ``stdout`` but not ``stderr``.
+
+Here is an example:
+
+.. code-block:: python
+
+    import taichi as ti
+
+    ti.init()
+    ti.set_logging_level(ti.ERROR)
+
+    try:
+        ti.error("It's a big error! Now exit...")
+    except RuntimeError as err:
+        print(err)
+
+Profiler
+--------
+
+Taichi profiler can help you count the time cost of your program. There are 2 profilers in Taichi available at this moment: ``ScopedProfiler`` and ``ProfilerBase``.
+
+ScopedProfiler
+##############
+
+1. ``ScopedProfiler`` evaluates the time spent for each step of the **host tasks** during execution hierarchically.
+
+2. Moreover, this profiler is turned on automatically and no extra work is needed except writing ``ti.core.print_profile_info()`` at the end of your Taichi program. For example:
+
+.. code-block:: python
+
+    ti.init(arch = ti.cpu)
+    var = ti.var(ti.f32, shape = 1)
+
+    @ti.kernel
+    def compute():
+        var[0] = 1.0
+        print(f" set var[0] = {var[0]}")
+
+    compute()
+    ti.core.print_profile_info()
+
+``ti.core.print_profile_info()`` will output statistics in a hierarchical format, with different colors for different levels.
```

Comment:
OFT: Can we make this ti.print_profile_info?

Reply:
Done, update comming soon.